### PR TITLE
[AIRFLOW-3264] URL decoding when parsing URI for connection

### PR DESCRIPTION
**Pull request by potiuk**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
  - https://issues.apache.org/jira/browse/AIRFLOW-3264

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Full URL decoding is performed now when parsing different
components of URI for connection. This enables to configure
paths to sockets including (for example ":") - so far
only '/' (%2f) was hard-coded in hostname. This change introduces
full decoding for all components of the URI.

Note that this is potentially breaking change if someone uses
% in some of their AIRFLOW_CONN_ defined connections.

### Tests

- [x] My PR adds the following unit tests :
tests.models.ConnectionTest:
  - test_connection_from_uri_with_colon_in_hostname
  - test_connection_from_uri_with_encoded_password
  - test_connection_from_uri_with_encoded_user
  - test_connection_from_uri_with_encoded_schema
  - test_connection_from_uri_with_no_schema

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

No new functionality.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/models.py [hunk 1]
```diff
@@ -808,6 +808,17 @@ def get_hook(self):
...
 
+    def debug_info(self):
+        return ("id: {}. Host: {}, Port: {}, Schema: {}, "
+                "Login: {}, Password: {}, extra: {}".
+                format(self.conn_id,
+                       self.host,
+                       self.port,
+                       self.schema,
+                       self.login,
+                       "XXXXXXXX" if self.password else None,
```
Comment (Fokko) [inline airflow/models.py:822]: Nice one

**Review by Fokko — APPROVED** [review]
LGTM. Nice one @potiuk

**Comment by Fokko** [comment]
The PR looks conceptually good to me. @potiuk Can you fix the CI?

**Comment by potiuk** [comment]
@Fokko - Fixed! It's good to have such comprehensive test suite! It actually caught an error with no password connections.

**Comment by jghoman** [comment]
Thanks.  Merging.

**Pull request merged**